subprojects/base-services/src/main/java/org/gradle/internal/util/PropertiesUtils.java

```diff
@@ -80,7 +79,11 @@ public static void store(Properties properties, OutputStream outputStream, @Null
         }
 
         Collections.sort(nonCommentLines);
-        String contents = Joiner.on(lineSeparator).join(lines);
-        outputStream.write(contents.getBytes(charset));
+        StringBuilder builder = new StringBuilder();
+        for (String line : lines) {
+            builder.append(line);
+            builder.append(lineSeparator);
+        }
+        outputStream.write(builder.toString().getBytes(charset));
```

Comment:
Why do it like this?

Reply:
So we have the trailing `\n`, as written by `Properties.store`. Joiner.on() only adds in between, or am I missing something?

Reply:
OK.